src/raster_types.jl

```diff
@@ -12,7 +13,24 @@ function Raster(dataset::GDALDatasetH)
                     _getrastercount(dataset)) # number of bands
     finalizer(raster, closeraster)
     raster
-end 
+end
+
+function Base.show(io::IO,raster::Raster)
+     if (raster.dataset == C_NULL)
+         println(io, raster.dataset)
+         return null;
```

Comment:
This is returning a function (making the return type unstable). Rather than explicitly returning `nothing`, either move it into the main if block or `return print...`.